test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/EverythingOperationTests.cs

```diff
@@ -0,0 +1,152 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Net;
+using Hl7.Fhir.Model;
+using Microsoft.Health.Fhir.Client;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Microsoft.Health.Fhir.Tests.E2E.Rest.Search;
+using Microsoft.Health.Test.Utilities;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Tests.E2E.Rest
+{
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.Json)]
+    public class EverythingOperationTests : SearchTestsBase<EverythingOperationTestFixture>
+    {
+        public EverythingOperationTests(EverythingOperationTestFixture fixture)
+            : base(fixture)
+        {
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.Json)]
+        public async Task GivenAPatientEverythingOperationWithId_WhenSearched_ThenResourcesInScopeShouldBeReturned()
+        {
+            string searchUrl = $"Patient/{Fixture.Patient.Id}/$everything";
+
+            await ExecuteAndValidateBundle(searchUrl, true, 4, Fixture.Patient, Fixture.Organization, Fixture.Device, Fixture.Observation, Fixture.Appointment, Fixture.Encounter);
+        }
+
+        [Fact]
+        [Trait(Traits.Priority, Priority.One)]
+        [HttpIntegrationFixtureArgumentSets(DataStore.CosmosDb, Format.Json)]
+        public async Task GivenAPatientEverythingOperationWithNonExistentId_WhenSearched_ThenResourcesInScopeShouldBeReturned()
+        {
+            string searchUrl = $"Patient/{Fixture.NonExistentPatient.Id}/$everything";
```

Comment:
>Fixture.NonExistentPatient.Id [](http://example.com/codeflow?start=42&length=29)

[https://www.hl7.org/fhir/patient-operation-everything.html](https://www.hl7.org/fhir/patient-operation-everything.html)

>At a minimum, the patient resource(s) itself is returned
>The intended use for this operation is to provide a patient with access to their entire record

From what I'm reading in operation definition we should return `Not found` here.

Reply:
Based on customer requirements, I'd like the data to still be returned. Cigna currently doesn't load the patient in the database but wants to get the data related to the patient (they have a placeholder id).

I think the question is more around what if the patient actually doesn't exist. How is that handled?

Reply:
There are 3 kinds of "non-existent" cases: (1) The Patient resource is not created, and its ID is not referenced in other resources. In this case, currently a Bundle with empty entry is returned. (2) The Patient resource is not created, but its ID has been referenced in other resources. (I think Cigna belongs to this case.) In this case, currently the Device resources and compartment resources are searched. (3) The Patient resource has been created, but is now deleted. The behavior is the same as (2).